sensors/sensors.go

```diff
@@ -0,0 +1,33 @@
+// Package sensors implements the high level structure of how sensors
+// are defined in slam libraries
+package sensors
+
+import (
+	"github.com/pkg/errors"
+
+	"go.viam.com/slam/config"
+)
+
+// Sensor contains the necessary information to define a generic sensor.
+type Sensor struct {
+	ID                string
```

Comment:
Do the individual Sensor types (Depth, etc) need to set ID, Index, and DefaultDataRateMs?